src/GraphQL/Types/Collections/SchemaTypes.cs

```diff
@@ -501,6 +535,53 @@ private void AddTypeIfNotRegistered(IGraphType type, TypeCollectionContext conte
             }
         }
 
+        private object RebuildType(Type type, bool input, List<(Type, Type)> typeMappings)
+        {
+            if (!type.IsGenericType)
+                return type;
+
+            var genericDef = type.GetGenericTypeDefinition();
+            if (genericDef == typeof(NonNullGraphType<>) || genericDef == typeof(ListGraphType<>))
+            {
+                var innerType = type.GenericTypeArguments[0];
+                object typeOrError = RebuildType(innerType, input, typeMappings);
+                if (typeOrError is string)
+                    return typeOrError;
+                var changed = (Type)typeOrError;
+                return changed == innerType ? type : genericDef.MakeGenericType(changed);
+            }
+            else if (genericDef == typeof(GraphQLClrOutputTypeReference<>) || genericDef == typeof(GraphQLClrInputTypeReference<>))
+            {
+                return GetGraphType(type.GetGenericArguments()[0], input, typeMappings);
+            }
+            else
+            {
+                return type;
+            }
+        }
+
+        private object GetGraphType(Type clrType, bool input, List<(Type clr, Type graph)> typeMappings)
```

Comment:
ditto

Reply:
The situation is the same here, because the result is simply passed up to `RebuildType` and further.